src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts

```diff
@@ -415,16 +422,53 @@ export class ObjectExplorerService implements IObjectExplorerService {
 						nodeProviders = nodeProviders.sort((a, b) => a.group!.toLowerCase().localeCompare(b.group!.toLowerCase()));
 						allProviders.push(...nodeProviders);
 					}
+					this.logService.trace(`${session.sessionId}: got providers for node expansion: ${allProviders.map(p => p.providerId).join(', ')}`);
+
+					const resolveExpansion = () => {
+						resolve(self.mergeResults(allProviders, resultMap, node.nodePath));
+						// Have to delete it after get all responses otherwise couldn't find session for not the first response
+						clearTimeout(expansionTimeout);
+						if (newRequest) {
+							delete self._sessions[session.sessionId!].nodes[node.nodePath];
+							this.logService.trace(`Deleted node ${node.nodePath} from session ${session.sessionId}`);
```

Comment:
We should definitely be avoiding mixing self and this - at best it isn't needed and at worst it indicates a bug. But since this is a lambda function you shouldn't need to use self so should be able to remove it for this and other similar functions.